src/include/OSL/variant.h

```diff
@@ -132,99 +132,93 @@ template<typename TBuiltinArg> class ArgVariant {
         return false;
     }
 
-    template<typename T> T get() const
-    {
-        OSL_DASSERT(false);
-        return T();
-    }
-
-    template<> inline TBuiltinArg get() const
+    TBuiltinArg get_builtin() const
     {
         OSL_DASSERT(is_holding<TBuiltinArg>());
         return m_builtin;
     }
 
-    template<> inline bool get() const
+    bool get_bool() const
```

Comment:
Any thoughts on this API moving away from std::variant?

Reply:
I'm wondering if we're mostly reinventing the TypeDesc and ParamValue types from OIIO here.

Reply:
Maybe, this was a bit of a clean room design and Lewis noted the needs lined up with std::variant pretty well.  ParamValue is perhaps a superset (has more functionality than we need) of most functionality, one major difference is ParamValue has pointer to value vs. nested value embedded in the type.  I think the need for the TBuiltinArg may be a deal breaker for trying to reuse ParamValue though.

Reply:
Hopefully it hasn't moved too far away from `std::variant`, the intention is still to switch once we get C++17. It will just be a small change when we do, e.g. `arg.get_bool()` -> `std::get<bool>(arg)`.

I just had to slightly change the interface as having template specialisations for the class and the member function was difficult and was not playing nice with all CI compilers consistently.

Reply:
Regarding TypeDesc/ParamValue, indeed one main reason to not use them here is the need to support TBuiltinArg. 

The other reason is to have as much compile time type checking as possible. E.g. currently you can't pass a Vec3/Mat44/etc. through, whereas you could construct a TypeDesc/ParamValue with one, then the LLVM gen would fail unless it supports everything.

Since it's easy to get into positions where the LLVM gen fails due to type incompatibilities (and it doesn't print a lot of helpful info for debugging) I wanted to try to reduce that potential as much as possible.

Reply:
Yeah, don't worry about that. I was just noticing the conceptual resemblance and musing about whether we might be needlessly reinventing something new. But I am convinced that you are doing the right thing.